src/CommandLine/Text/HelpText.cs

```diff
@@ -269,11 +269,13 @@ public SentenceBuilder SentenceBuilder
 
             var errors = Enumerable.Empty<Error>();
 
+         
             if (onError != null && parserResult.Tag == ParserResultType.NotParsed)
             {
                 errors = ((NotParsed<T>)parserResult).Errors;
-
-                if (errors.OnlyMeaningfulOnes().Any())
+                if (errors.IsHelp())
+                    auto = onError(auto);
+                else if (errors.OnlyMeaningfulOnes().Any())

```

Comment:
I suggest simplifying: 
```csharp
if (errors.IsHelp() || errors.OnlyMeaningfulOnes().Any())
   auto = onError(auto);

Reply:
Thanks for your suggestions.
I'll modify my code to use yours.